assets/client.go

```diff
@@ -184,6 +185,67 @@ func (c *TapdClient) GetAssetName(ctx context.Context,
 	return assetName, nil
 }
 
+// GetAssetPrice returns the price of an asset in satoshis. NOTE: this currently
+// uses the rfq process for the asset price. A future implementation should
+// use a price oracle to not spam a peer.
+func (c *TapdClient) GetAssetPrice(ctx context.Context, assetID string,
+	peerPubkey []byte, assetAmt, paymentMaxAmt uint64) (btcutil.Amount,
+	error) {
+
+	// We'll allow a short rfq expiry as we'll only use this rfq to
+	// gauge a price.
+	rfqExpiry := time.Now().Add(time.Minute).Unix()
+
+	// First we'll rfq a random peer for the asset.
+	rfq, err := c.RfqClient.AddAssetSellOrder(
+		ctx, &rfqrpc.AddAssetSellOrderRequest{
+			AssetSpecifier: &rfqrpc.AssetSpecifier{
+				Id: &rfqrpc.AssetSpecifier_AssetIdStr{
+					AssetIdStr: assetID,
+				},
+			},
+			PaymentMaxAmt:  paymentMaxAmt,
+			Expiry:         uint64(rfqExpiry),
+			TimeoutSeconds: uint32(c.cfg.RFQtimeout.Seconds()),
+			PeerPubKey:     peerPubkey,
+		})
+	if err != nil {
+		return 0, err
+	}
+	if rfq.GetInvalidQuote() != nil {
+		return 0, fmt.Errorf("invalid RFQ: %v", rfq.GetInvalidQuote())
+	}
+
+	if rfq.GetRejectedQuote() != nil {
+		return 0, fmt.Errorf("rejected RFQ: %v", rfq.GetRejectedQuote())
+	}
+
+	acceptedRes := rfq.GetAcceptedQuote()
+	if acceptedRes == nil {
+		return 0, fmt.Errorf("no accepted quote")
+	}
+
+	// We'll use the accepted quote to calculate the price.
+	return getSatsFromAssetAmt(assetAmt, acceptedRes.BidAssetRate)
+}
+
+// getSatsFromAssetAmt returns the amount in satoshis for the given asset amount
+// and asset rate.
+func getSatsFromAssetAmt(assetAmt uint64, assetRate *rfqrpc.FixedPoint) (
+	btcutil.Amount, error) {
+
+	rateFP, err := rfqrpc.UnmarshalFixedPoint(assetRate)
+	if err != nil {
+		return 0, fmt.Errorf("cannot unmarshal asset rate: %w", err)
+	}
+
+	assetUnits := rfqmath.NewBigIntFixedPoint(assetAmt, 0)
```

Comment:
Here we hold the assumption that an asset unit is at least a millisatoshi. What about fractions? Should we scale both asset units and the rate so we can represent fractions correctly?